# fix for BiDi by Ilya Konstantinov <future@shiny.co.il>
--- src/fe-gtk/xtext.c	23 May 2004 06:46:34 -0000	1.93
+++ src/fe-gtk/xtext.c	5 Jun 2004 05:36:22 -0000
@@ -2826,28 +2826,25 @@
 		if (i > len)
 			i = len;
 
-		/* separate the left part, the space and the right part
-		   into separate runs, and reset bidi state inbetween */
-
-/* ... left string, the space and the right
-string of a text entry are being drawn in a separate runs.
-This ensures that the space never joins the 'right' part of the text
-entry, which proves to be a problem when the text entry starts with a
-Right-to-Left character (since then, the space, which is intended for
-the separator, joins the RTL character to the right of it).*/
-
-		/* we've reached the end of the left part? */
-		if ((pstr-str)+j == ent->left_len)
-		{
-			x += gtk_xtext_render_flush (xtext, x, y, pstr, j, gc, ent->mb);
-			pstr += j;
-			j = 0;
-		}
-		else if ((pstr-str)+j == ent->left_len+1)
+		/* Separate the left part, the space and the right part
+		   into separate runs, and reset bidi state inbetween.
+		   Perform this only on the first line of the message.
+                */
+		if (offset == 0)
 		{
-			x += gtk_xtext_render_flush (xtext, x, y, pstr, j, gc, ent->mb);
-			pstr += j;
-			j = 0;
+			/* we've reached the end of the left part? */
+			if ((pstr-str)+j == ent->left_len)
+			{
+				x += gtk_xtext_render_flush (xtext, x, y, pstr, j, gc, ent->mb);
+				pstr += j;
+				j = 0;
+			}
+			else if ((pstr-str)+j == ent->left_len+1)
+			{
+				x += gtk_xtext_render_flush (xtext, x, y, pstr, j, gc, ent->mb);
+				pstr += j;
+				j = 0;
+			}
 		}
 
 		/* have we been told to stop rendering at this point? */
